### PR TITLE
fix connection where with logical operator

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Bug-fix Connection where with logical operator

Fix a bug introduced with the `NOT` operator PR, logical filters inside a connection field was treated as an array, but the `NOT` operator value is an object. 

## Complexity

Complexity: Literally one change

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2807.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Review by Liam-Doodson — DISMISSED** [review]
LGTM!

File: packages/graphql/src/translate/where/property-operations/create-connection-operation.ts [hunk 1]
```diff
@@ -197,6 +197,7 @@ export function createConnectionWherePropertyOperation({
     Object.entries(whereInput).forEach(([key, value]) => {
         if (isLogicalOperator(key)) {
             const subOperations: (Cypher.Predicate | undefined)[] = [];
+            value = Array.isArray(value) ? value : [value];
```
Comment (angrykoala) [inline packages/graphql/src/translate/where/property-operations/create-connection-operation.ts:200]: Consider using `asArray` here, it should also fix the typings so there is no need to cast `asArray<any>`

**Comment by darrellwarde** [comment]
Changeset? 👀

**Comment by MacondoExpress** [comment]
@darrellwarde  I believe the NOT operator is not released yet, do we need the Changeset even for internal changes not yet released?

File: packages/graphql/src/translate/create-aggregate-where-and-params.ts [hunk 2]
```diff
@@ -139,8 +139,7 @@ function aggregateWhere(
         } else if (isLogicalOperator(key)) {
             const cypherBuilderFunction = getCypherLogicalOperator(key);
             const logicalPredicates: Cypher.Predicate[] = [];
-            value = Array.isArray(value) ? value : [value];
-            value.forEach((whereInput) => {
+            (Array.isArray(value) ? value : [value]).forEach((whereInput) => {
```
Comment (a-alle) [inline packages/graphql/src/translate/create-aggregate-where-and-params.ts:142]: there's an `asArray` utility function in `src/utils/utils.ts`

**Comment by darrellwarde** [comment]
> @darrellwarde I believe the NOT operator is not released yet, do we need the Changeset even for internal changes not yet released?

Perhaps not!!

**Pull request merged**